### DISCUSSION TITLE
Conditional Mandatory Rule for BT-109-Lot does not function reliably

### DISCUSSION THREAD

**Discussion by lesomue** [OP-TED/eForms-SDK#929] [Q&A] [answered]
Hello everyone, 

During the implementation of SDK 1.10.1 we stumbled across the fact that BR-BT-00109-0023 does not always trigger, although the conditions for the error according to fields.json are fulfilled. 

Specifically, the fields.json for notice subtype 18, for example, states BT-109 is mandatory if it is not forbidden and that it is only forbidden if the duration is shorter than 8 years. 

If you now enter a duration of 10 years and leave BT-109 empty, the rule does not fire. If BT-113 is also entered, the rule is triggered, although the rule should be independent from BT-113. 

We suspect that this is related to the fact that the rule context for BR-BT-00109-0023
`<rulecontext="/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingProcess/cac:FrameworkAgreement[$noticeSubType = '17']" `
is only present if BT-113 is used. If not, the rule context is not present and the rule is not triggered, because there is no xpath to validate. 

That means, a contracting authority only knows they are supposed to enter data for BT-109 when the duration exceeds 8 years **AND** they decide to enter BT-113. Otherwise, they would not receive a validation error and would not know that they violate a rule stipulated in the fields.json. 

Could you please tell us how to proceed in this case? 

Many thanks in advance and best regards 
Lea Müller 
cosinex GmbH

**Comment by YvesJo** [comment, marked as answer]
That's correct, the context will exist only if BT-109, BT-111 or BT-113 exists. Context will et updated in SDK 1.12
